src/app/app.component.ts

```diff
@@ -21,23 +21,13 @@ export class AppComponent implements OnInit {
   ) { }
 
   ngOnInit(): void {
-    // We must be in an iframe OR opened with window.open
-    if (!this.globalVars.inTab && !this.globalVars.inFrame()) {
-      window.location.href = `https://${this.globalVars.environment.nodeHostname}`;
-      return;
-    }
-
-    this.identityService.initialize().subscribe(res => {
-      this.globalVars.hostname = res.hostname;
-      if (this.globalVars.isFullAccessHostname()) {
-        this.globalVars.accessLevelRequest = AccessLevel.Full;
-      }
-
-      this.loading = false;
-    });
-
-    // Store testnet for duration of this session
     this.activatedRoute.queryParams.subscribe(params => {
+      // We must be in an iframe OR opened with window.open OR running in a webview
+      if (!params.webview && !this.globalVars.inTab && !this.globalVars.inFrame()) {
```

Comment:
Did you test this locally? Angular is very annoying and fires queryParams twice and the first event is always empty. This might redirect regardless of the webview param?

Reply:
Yeah, I tested it and it worked fine. If I pass webview=true it didn't redirect. If I left it off it redirected, so I think it's fine